lib/workers/bin/run_single_worker.rb

```diff
@@ -60,9 +64,9 @@
   create_options = {:pid => Process.pid}
   runner_options = {}
 
-  if ENV["QUEUE"]
-    create_options[:queue_name] = ENV["QUEUE"]
-    runner_options[:ems_id] = worker_class.ems_id_from_queue_name(ENV["QUEUE"]) if worker_class.respond_to?(:ems_id_from_queue_name)
+  if options[:ems_id]
+    create_options[:queue_name] = options[:ems_id].length == 1 ? "ems_#{options[:ems_id].first}" : options[:ems_id].collect { |id| "ems_#{id}" }
+    runner_options[:ems_id]     = options[:ems_id].length == 1 ? options[:ems_id].first : options[:ems_id].collect { |id| id }
   end
```

Comment:
I really don't like how I did this. The problem is, `create_options[:queue_name]` & `runner_options[:ems_id]` can accept an array when there are multiple `ems ids`. However when there is just a singular `ems id`. I don't think that works as well. So I am trying to pass a single `ems id`, but it just makes this look a mess.

Any suggestions to make it look nicer?

Reply:
Let's try to find a better way to do it.  I can't think of one right now.  Your changes are great for now.  We can clean this up when we find a better place for this logic.